server/app/mutations/grid_services/update.rb

```diff
@@ -40,11 +41,11 @@ def validate
             !vols.include?(v)
           }
           if changed_volumes.any? { |v| !v.include?(':') }
-            add_error(:volumes, :invalid, 'Adding a non-named volume is not supported to a stateful service')
+            add_error(:volumes, :invalid, 'Adding a non-named volume to a stateful service is not supported')
```

Comment:
This is sadly inconsistent still, the error here is like `{volumes: "Adding ..." }`, but the `validate_volumes` may override that with an array-shaped error like `{volumes: [ ... ] }`.